src/python/grpcio_tests/tests_aio/unit/client_unary_stream_interceptor_test.py

```diff
@@ -0,0 +1,404 @@
+        self.assertTrue(call.done())
+        self.assertEqual(await call.code(), grpc.StatusCode.CANCELLED)
+        self.assertEqual(await call.details(),
+                         _LOCAL_CANCEL_DETAILS_EXPECTATION)
+        self.assertEqual(await call.initial_metadata(), None)
+        self.assertEqual(await call.trailing_metadata(), None)
+        await channel.close()
+
+    async def test_cancel_after_rpc(self):
+
+        interceptor_reached = asyncio.Event()
+        wait_for_ever = self.loop.create_future()
+
+        class Interceptor(aio.UnaryStreamClientInterceptor):
+
+            async def intercept_unary_stream(self, continuation,
+                                             client_call_details, request):
+                call = await continuation(client_call_details, request)
+                interceptor_reached.set()
+                await wait_for_ever
+
+        channel = aio.insecure_channel(UNREACHABLE_TARGET,
+                                       interceptors=[Interceptor()])
+        request = messages_pb2.StreamingOutputCallRequest()
+        stub = test_pb2_grpc.TestServiceStub(channel)
+        call = stub.StreamingOutputCall(request)
+
+        self.assertFalse(call.cancelled())
+        self.assertFalse(call.done())
+
+        await interceptor_reached.wait()
+        self.assertTrue(call.cancel())
+
+        with self.assertRaises(asyncio.CancelledError):
+            async for response in call:
+                pass
+
+        self.assertTrue(call.cancelled())
+        self.assertTrue(call.done())
+        self.assertEqual(await call.code(), grpc.StatusCode.CANCELLED)
+        self.assertEqual(await call.details(),
+                         _LOCAL_CANCEL_DETAILS_EXPECTATION)
+        self.assertEqual(await call.initial_metadata(), None)
+        self.assertEqual(await call.trailing_metadata(), None)
+        await channel.close()
+
+    async def test_cancel_consuming_response_iterator(self):
+        request = messages_pb2.StreamingOutputCallRequest()
+        for _ in range(_NUM_STREAM_RESPONSES):
+            request.response_parameters.append(
+                messages_pb2.ResponseParameters(
+                    size=_RESPONSE_PAYLOAD_SIZE,
+                    interval_us=_RESPONSE_INTERVAL_US))
+
+        channel = aio.insecure_channel(
+            self._server_target,
+            interceptors=[_UnaryStreamInterceptorWith_ResponseIterator()])
+        stub = test_pb2_grpc.TestServiceStub(channel)
+        call = stub.StreamingOutputCall(request)
+
+        with self.assertRaises(asyncio.CancelledError):
+            async for response in call:
+                call.cancel()
+
+        self.assertTrue(call.cancelled())
+        self.assertTrue(call.done())
+        self.assertEqual(await call.code(), grpc.StatusCode.CANCELLED)
+        self.assertEqual(await call.details(),
+                         _LOCAL_CANCEL_DETAILS_EXPECTATION)
+        await channel.close()
```

Comment:
Can we add test cases
1. The interceptor cancels the entire call within `intercept_unary_stream` (to ensure the call functionality is working);
2. An exception is raised within `intercept_unary_stream` (so we fix the error path behavior).

Reply:
WDYM by

> (so we fix the error path behavior)

If an Exception is raised this Exception will be propagated later on and raise during the iterator, and we will do anything special since is an `Exception` that its not even a `Cancellation` or an `AioRpcError`

Reply:
I mean make sure we don't change the error path behavior. It is defined by us. I made a bad word choice.